File: src/lib/shared/types/Playlist.ts
import type {ExternalUrls} from "./ExternalUrls";
import type {Followers} from "./Followers";
import type {Image} from "./Image";

export interface Playlist {
  collaborative?: boolean
  description?: string | null
  external_urls?: ExternalUrls
  followers?: Followers
  href: string
  id: string
  images?: Image[]
  name: string
  owner: { id: string, name: string }
  public?: boolean
  snapshot_id: string
  tracks: { id: string, name: string }[]
  type?: string
  uri?: string
}

File: lib/api/schemas/playlist.ex
defmodule Playlist do
  use Api.Schema

  schema "playlists" do
    field :collaborative, :boolean
    field :description, :string
    embeds_one :external_urls, ExternalUrls
    embeds_one :followers, Followers
    field :href, :string
    field :spotify_id, :string
    embeds_many :images, Image
    field :name, :string
    embeds_one :owner, SimpleUser
    field :public, :boolean
    field :snapshot_id, :string
    field :total_tracks, :integer
    has_many :tracks, PlaylistTrack, foreign_key: :id
    field :type, Ecto.Enum, values: [:playlist]
    field :uri, :string
  end
end
